src/org/rascalmpl/uri/classloaders/SourceLocationClassLoader.java

```diff
@@ -40,7 +42,7 @@
 public class SourceLocationClassLoader extends ClassLoader {
     private final Map<String, Class<?>> cache = new ConcurrentHashMap<>();
     private final List<ClassLoader> path;
-    private final Stack<SearchItem> stack = new Stack<>();
+    private final ThreadLocal<Deque<SearchItem>> stack = ThreadLocal.withInitial(ArrayDeque::new);
```

Comment:
One of the issues with `ThreadLocal` is that they're never cleared. So I think instead of `ArrayDeque` we should use a `LinkedList` (which also implements `Deque`) just so that we don't leave around big arrays's that never got cleared.

Reply:
Ah yes. Agreed!